jac-uk/apply
# Prevent sign up with 'judicialappointments.gov.uk' emails

## What's included?
When an admin uses their gov.uk account to sign up as a candidate, it blocks them from signing in as an admin.
This happens quite often and has become a regular maintenance task. 
There are firebase configurations which would allow multiple accounts to be held under the same email address, but that presents additional issues.

Instead this code uses regex to match the email field and present a validation error when '@judicialappointments.gov.uk' is detected. 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
- Sign out.
- Go to [sign up page](https://jac-apply-develop--pr820-prevent-signup-with-13gj3knq.web.app/sign-up)
- type '[ANY]@judicialappointments.gov.uk'
- attempt to sign up
- observe validation error

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work
🟠 Medium risk - this does change code that is shared with other areas
🔴 High risk - this includes a lot of changes to shared code

## Additional context
![image](https://user-images.githubusercontent.com/44227249/146047083-a8ee36f1-8337-4975-83bd-29b7aec1a92b.png)
---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
